Microsoft.Toolkit.Uwp.UI.Controls/FlexPanel/FlexPanel.cs

```diff
@@ -0,0 +1,1613 @@
+        /// elements with a FlexBasis.IsRelstive set to true.
+        /// </summary>
+        /// <remarks>The default value for this property is Auto.</remarks>
+        /// <returns>FlexBasis</returns>
+        public static FlexBasis GetFlexBasis(UIElement element)
+        {
+            if (GetBasis(element) is string basisString)
+            {
+                return FlexBasis.Parse(basisString);
+            }
+
+            return FlexBasis.Auto;
+        }
+
+        private static readonly DependencyProperty FlexItemProperty = DependencyProperty.RegisterAttached(
+            "FlexItem",
+            typeof(object),
+            typeof(FlexPanel),
+            new PropertyMetadata(null));
+
+        private static FlexItem GetFlexItem(UIElement element)
+        {
+            if (element is null)
+            {
+                return null;
+            }
+
+            if (element is FlexPanel flexPanel)
+            {
+                return flexPanel._root;
+            }
+
+            FlexItem item = null;
+            try
+            {
+                item = (FlexItem)element.GetValue(FlexItemProperty);
+            }
+            catch (Exception)
+            {
+            }
+
+            if (item is null)
+            {
+                item = new FlexItem();
+                element.SetValue(FlexItemProperty, item);
+            }
+
+            return item;
+        }
+
+        private static void SetFlexItem(UIElement element, FlexItem value)
+        {
+            element.SetNewValue(FlexItemProperty, value);
+            UpdateItemProperties(element, value);
+        }
+
+        private static void InternalInvalidateArrange(UIElement element)
+        {
+            if (element is FlexPanel)
+            {
+                element.InvalidateArrange();
+            }
+            else if (element is FrameworkElement frameworkElement && frameworkElement.Parent is FlexPanel flexPanel)
+            {
+                flexPanel.InvalidateArrange();
+            }
+        }
+
+        private static void UpdateItemProperties(UIElement view, FlexItem item)
+        {
+            item.IsVisible = view.Visibility == Visibility.Visible;
+
+            if (view is FrameworkElement element)
+            {
+                item.MarginLeft = (double)element.Margin.Left;
+                item.MarginTop = (double)element.Margin.Top;
+                item.MarginRight = (double)element.Margin.Right;
+                item.MarginBottom = (double)element.Margin.Bottom;
+            }
+
+            if (view is Control control)
+            {
+                item.SetPadding(control.Padding);
+            }
+            else if (view is Border border)
+            {
+                item.SetPadding(border.Padding);
+            }
+            else if (view is TextBlock textBlock)
+            {
+                item.SetPadding(textBlock.Padding);
+            }
+#if __ANDROID__ || __WASM__
```

Comment:
No, they are for Uno platform.   Probably not appropriate but, as you can see, the changes are needed to get this to work with Uno Android and WebAssembly.  

This is a good chance to allow me at share why I've submitted this PR for consideration.  I have developed a number of Xamarin.Forms apps for the construction and engineering segment.  It's a long story but I believe I am painted into a corner with Xamarin.Forms.  Specifically very poor performance on Android and the lack of a Web Assembly support (UWP and iOS performance is great!).  Uno has the Web Assembly and I have some evidence that suggests they have worked around the Android performance issues. With that in mind, I am in the process of porting a large library of user interface code to Uno.   Flex panel is one of the more complex controls I need and I thought it would be beneficial to share - and I'm learning a lot from your great feedback.

So this raises an important question.  At the most recent Uno Conference, there was some indication that there is some cooperation between the Uno Team and the WCT team to extent WCT to Uno.  I guess I've front loaded that here but don't know if that's appropriate or helpful.  Would you be willing to share with me an impression of how this cooperation works and how Uno specific changes are integrated into WCT code?

Reply:
I can't talk about this myself (I'm not part of the team so I don't know all the details on this), but I can defer to @michael-hawker to comment on this specific point, he'll definitely be able to answer your questions here 😊

My issue here was mostly just about having those compiler directives directly in the toolkit, which we can't test ourselves anyway since the package is only built for UWP. The Uno guys have their own fork of the toolkit, so I think checks/conditions like those should just go into their version of the codebase.

Reply:
Yeah, the Uno team has their own fork at the moment. @azchohfi @Sergio0694 is there harm in leaving these here to be picked up later downstream? As Sergio calls out it's not something we currently test though.

Reply:
@michael-hawker There shouldn't be any harm, those constants would just never be defined so the code in those branches would just never get compiler at all. It would mostly just be about "clutter" in the codebase, but there's no actual harm as far as the final assembly is concerned, that's not a concern 😄